apps/database/src/models/Account.ts

```diff
@@ -131,19 +131,17 @@ export class Account extends Model {
     }
 
     const activePurchase = await this.getActivePurchase();
-
     if (!activePurchase || !activePurchase.startDate) {
       return startOfMonth;
     }
 
-    const purchaseStartDate = new Date(activePurchase.startDate);
-
-    const startOfSecondMonth = new Date(
-      purchaseStartDate.getFullYear(),
-      purchaseStartDate.getMonth() + 1,
-      1
+    const purchaseDay = new Date(activePurchase.startDate).getDate();
+    const hasReset = now.getDate() >= purchaseDay;
+    return new Date(
+      now.getFullYear(),
+      hasReset ? now.getMonth() : now.getMonth() - 1,
+      purchaseDay
     );
```

Comment:
It will result to an invalid date if the `purchaseDay` is 31.

The correct formula is:

```js
const now = new Date()
const purchaseDate = new Date(activePurchase.startDate)
const startOfPurchaseMonth = startOfMonth(purchaseDate)
const startPeriodAfterMs = startOfPurchaseMonth.getTime() - purchaseDate.getTime();
const startOfCurrentMonth = startOfMonth(now)
const endOfCurrentMonth = endOfMonth(now)
const startOfCurrentPeriod = min(endOfCurrentMonth, new Date(startOfCurrentMonth + startPeriodAfterMs))
```

Functions comes from [date-fns](https://date-fns.org/).

Also you don't take care of time, your date is approximative. It means the number displayed on Stripe and on Argos will not be the same.